### PR TITLE
OGC API support

### PR DESCRIPTION
**Pull request by lucadelu**
This two addons are first draft of modules to import data from OGC API Features and OGC API Coverages standards.

I created r.in.ogc and v.in.ogc directory because other standards will be added:
- for raster OGC API Maps and OGC API Tiles
- for vector OGC API EDR

### CODE REVIEW

File: grass7/raster/r.in.ogc/r.in.ogc.coverages/r.in.ogc.coverages.html [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<h2>DESCRIPTION</h2>
+<em>r.in.ogc.coverages</em> in able to import data from a OGC API Coverages Service
```
Comment (veroandreo) [inline grass7/raster/r.in.ogc/r.in.ogc.coverages/r.in.ogc.coverages.html:2]:
```suggestion
<em>r.in.ogc.coverages</em> is able to import data from an OGC API Coverages Service
```

File: grass7/raster/r.in.ogc/r.in.ogc.coverages/r.in.ogc.coverages.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+#% description: Get layers from the server
+#%end
+import sys
+import grass.script as grass
+
+def main():
+    try:
+        from owslib.ogcapi.coverages import Coverages
+    except:
+        grass.fatal(_("OSWLib can not be found. Install OSWLib (http://geopython.github.com/OWSLib/)."))
```
Comment (veroandreo) [inline grass7/raster/r.in.ogc/r.in.ogc.coverages/r.in.ogc.coverages.py:52]:
```suggestion
        grass.fatal(_("OSWLib was not found. Install OSWLib (http://geopython.github.com/OWSLib/)."))
```

File: grass7/raster/r.in.ogc/r.in.ogc.html [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+<link rel="stylesheet" href="grassdocs.css" type="text/css">
+</head>
+<body bgcolor="white">
+<div id="container">
+
+<a href="https://grass.osgeo.org/grass-stable/manuals/index.html"><img src="grass_logo.png" alt="GRASS logo"></a>
+<hr class="header">
+
+<h2>NAME</h2>
+<em><b>r.in.ogc</b></em> import vector raster data from a OGC API server
```
Comment (veroandreo) [inline grass7/raster/r.in.ogc/r.in.ogc.html:15]:
```suggestion
<em><b>r.in.ogc</b></em> imports raster data from an OGC API server.
```

File: grass7/raster/r.in.ogc/r.in.ogc.html [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+<h2>KEYWORDS</h2>
+<a href="https://grass.osgeo.org/grass-stable/manuals/vector.html">vector</a>,
+<a href="https://grass.osgeo.org/grass-stable/manuals/topic_import.html">import topic</a>
+
+<h2>DESCRIPTION</h2>
+
+The <em>r.in.ogc</em> suite is able to import raster data from several OGC API standards. It support
```
Comment (veroandreo) [inline grass7/raster/r.in.ogc/r.in.ogc.html:22]:
```suggestion
The <em>r.in.ogc</em> suite is able to import raster data from several OGC API standards. It supports
```

File: grass7/raster/r.in.ogc/r.in.ogc.html [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+<ul>
+<li><a href="v.in.ogc.features.html">OGC API Coverages</a> - import raster data from a OGC API Coverages standard </li>
```
Comment (veroandreo) [inline grass7/raster/r.in.ogc/r.in.ogc.html:24]:
```suggestion
<li><a href="r.in.ogc.coverages.html">OGC API Coverages</a> - import raster data from an OGC API Coverages standard </li>
```

File: grass7/vector/v.in.ogc/v.in.ogc.features/v.in.ogc.features.html [hunk 6]
```diff
@@ -0,0 +1,18 @@
+<h2>DESCRIPTION</h2>
+<em>v.in.ogc.features</em> in able to import data from a OGC API Features Service
```
Comment (veroandreo) [inline grass7/vector/v.in.ogc/v.in.ogc.features/v.in.ogc.features.html:2]:
```suggestion
<em>v.in.ogc.features</em> is able to import data from an OGC API Features Service.
```

File: grass7/vector/v.in.ogc/v.in.ogc.features/v.in.ogc.features.py [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+import grass.script as grass
+
+
+def main():
+    try:
+        from owslib.ogcapi.features import Features
+    except:
+        grass.fatal(
+            _(
+                "OSWLib can not be found. Install OSWLib (http://geopython.github.com/OWSLib/)."
```
Comment (veroandreo) [inline grass7/vector/v.in.ogc/v.in.ogc.features/v.in.ogc.features.py:56]:
```suggestion
                "OSWLib was not found. Install OSWLib (http://geopython.github.com/OWSLib/)."
```

File: grass7/vector/v.in.ogc/v.in.ogc.features/v.in.ogc.features.py [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+    if not options["output"]:
+        grass.fatal(
+            _("Required parameter <output> not set: (Name for output vector map)")
+        )
+
+    try:
+        layer = feats.collection_items(options["layer"])
+    except Exception as e:
+        grass.fatal(
+            _("Problem retriving data from the server. The error was: {}".format(e))
```
Comment (veroandreo) [inline grass7/vector/v.in.ogc/v.in.ogc.features/v.in.ogc.features.py:81]:
```suggestion
            _("Problem retrieving data from the server. The error was: {}".format(e))
```

File: grass7/raster/r.in.ogc/r.in.ogc.coverages/r.in.ogc.coverages.py [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+
+    if not options["output"]:
+        grass.fatal(
+            _("Required parameter <output> not set: (Name for output vector map)")
+        )
+    try:
+        layer = feats.coverage(options["layer"])
+    except Exception as e:
+        grass.fatal(
+            _("Problem retriving data from the server. The error was: {}".format(e))
```
Comment (veroandreo) [inline grass7/raster/r.in.ogc/r.in.ogc.coverages/r.in.ogc.coverages.py:75]:
```suggestion
            _("Problem retrieving data from the server. The error was: {}".format(e))
```

File: grass7/vector/v.in.ogc/v.in.ogc.html [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+<link rel="stylesheet" href="grassdocs.css" type="text/css">
+</head>
+<body bgcolor="white">
+<div id="container">
+
+<a href="https://grass.osgeo.org/grass-stable/manuals/index.html"><img src="grass_logo.png" alt="GRASS logo"></a>
+<hr class="header">
+
+<h2>NAME</h2>
+<em><b>v.in.ogc</b></em> import vector data from a OGC API server
```
Comment (veroandreo) [inline grass7/vector/v.in.ogc/v.in.ogc.html:15]:
```suggestion
<em><b>v.in.ogc</b></em> imports vector data from an OGC API server.
```

File: grass7/vector/v.in.ogc/v.in.ogc.html [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+<h2>KEYWORDS</h2>
+<a href="https://grass.osgeo.org/grass-stable/manuals/vector.html">vector</a>,
+<a href="https://grass.osgeo.org/grass-stable/manuals/topic_import.html">import topic</a>
+
+<h2>DESCRIPTION</h2>
+
+The <em>v.in.ogc</em> suite is able to import vector data from several OGC API standards. It support
```
Comment (veroandreo) [inline grass7/vector/v.in.ogc/v.in.ogc.html:22]:
```suggestion
The <em>v.in.ogc</em> suite is able to import vector data from several OGC API standards. It supports
```

**Review by veroandreo — COMMENTED** [review]
minor typos fixed :)

**Comment by ninsbl** [comment]
This is great!
While it makes sense to start with addons, it would be great to get OGC API support into GRASS 8 core!

**Comment by lucadelu** [comment]
Yes could be useful but this means a new dependency (owslib)

**Pull request merged**